### PR TITLE
fix(levm): fix returndatacopy

### PR DESCRIPTION
**Pull request by JereSalo**
**Motivation**

**Description**

- Fix returndatacopy: If offset + size is larger than returndatasize then we have to revert execution, and we weren't doing that. [EVM Codes](https://www.evm.codes/?fork=cancun#3e)
- Changes name of `OutOfOffset` error for `OutOfBounds` because I consider it more appropriate.

Closes #issue_number

### CODE REVIEW

File: crates/vm/levm/src/errors.rs [hunk 1]
```diff
@@ -71,7 +71,7 @@ pub enum VMError {
     #[error("Transaction validation error: {0}")]
     TxValidation(#[from] TxValidationError),
     #[error("Offset out of bounds")]
-    OutOfOffset,
+    OutOfBounds,
```
Comment (lima-limon-inc) [inline crates/vm/levm/src/errors.rs:74]: Totally agree! "Out of offset" sounds a bit strage.

I was actually messing around with this same error in my branch!

**Review by lima-limon-inc — APPROVED** [review]
LGTM

File: crates/vm/levm/src/opcode_handlers/environment.rs [hunk 2]
```diff
@@ -385,10 +385,16 @@ impl VM {
...
+        let copy_limit = returndata_offset
+            .checked_add(size)
+            .ok_or(VMError::VeryLargeNumber)?;
+
+        if copy_limit > sub_return_data_len {
+            return Err(VMError::OutOfBounds);
         }
 
+        // Actually we don't need to fill with zeros for out of bounds bytes, this works but is overkill because of the previous validations.
+        // I would've used copy_from_slice but it can panic.
```
[the remark this answers is not in the document]
Comment (JereSalo) [reply crates/vm/levm/src/opcode_handlers/environment.rs:397]: They are mandatory in EOF code. We currently don't support that. You can read more about EOF [here](https://evmobjectformat.org/)

**Review by maximopalopoli — APPROVED** [review]
LGTM

File: crates/vm/levm/src/opcode_handlers/environment.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilitteri) [reply crates/vm/levm/src/opcode_handlers/environment.rs:397]: Yes, we'll take care about EOF soon, but we can avoid it for now.

**Review by ilitteri — APPROVED** [review]
LGTM

**Pull request merged**